README.md

```diff
@@ -128,17 +148,222 @@ To start Parity Ethereum as a regular user using `systemd` init:
+
+Example (generic):
+```markdown
+/// Summary
+///
+/// Description
+///
+/// # Panics
+///
+/// # Errors
+///
+/// # Safety
+///
+/// # Examples
+///
+/// Summary of Example 1
+///
+/// ```rust
+/// // insert example 1 code here
+/// ```
+///
+```
+
+* Important notes:
+  * Documentation comments must use annotations with a triple slash `///`
+  * Modules are documented using `//!`
+```
+//! Summary (of module)
+//!
+//! Description (of module)
+```
+* Special section header is indicated with a hash `#`.
+  * `Panics` section requires an explanation if the function triggers a panic
+  * `Errors` section is for describing conditions under which a function of method returns `Err(E)` if it returns a `Result<T, E>`
+  * `Safety` section requires an explanation if the function is `unsafe`
+  * `Examples` section includes examples of using the function or method
+* Code block annotations for examples are included between triple graves, as shown above.
+Instead of including the programming language to use for syntax highlighting as the annotation
+after the triple graves, alternative annotations include the `ignore`, `text`, `should_panic`, or `no_run`.
+* Summary sentence is a short high level single sentence of its functionality
+* Description paragraph is for details additional to the summary sentence
+* Missing documentation annotations may be used to identify where to generate warnings with `#![warn(missing_docs)]`
+or errors `#![deny(missing_docs)]`
+* Hide documentation for items with `#[doc(hidden)]`
+
+### Contributing to documentation (tests, extended examples, macros) for Parity Ethereum packages
+
+The code block annotations in the `# Example` section may be used as [documentation as tests and for extended examples](https://doc.rust-lang.org/1.9.0/book/documentation.html#documentation-as-tests).
+
+* Important notes:
+  * Rustdoc will automatically add a `main()` wrapper around the code block to test it
+  * [Documenting macros](https://doc.rust-lang.org/1.9.0/book/documentation.html#documenting-macros).
+  * Documentation as tests examples are included when running `cargo test`
+
+## 5. Toolchain <a id="chapter-005"></a>
 
 In addition to the Parity Ethereum client, there are additional tools in this repository available:
 
-- [evmbin](https://github.com/paritytech/parity-ethereum/blob/master/evmbin/) - EVM implementation for Parity Ethereum.
-- [ethabi](https://github.com/paritytech/ethabi) - Parity Ethereum function calls encoding.
-- [ethstore](https://github.com/paritytech/parity-ethereum/blob/master/accounts/ethstore) - Parity Ethereum key management.
-- [ethkey](https://github.com/paritytech/parity-ethereum/blob/master/accounts/ethkey) - Parity Ethereum keys generator.
-- [whisper](https://github.com/paritytech/parity-ethereum/blob/master/whisper/) - Implementation of Whisper-v2 PoC.
+- [evmbin](https://github.com/paritytech/parity-ethereum/blob/master/evmbin/) - Parity Ethereum EVM Implementation.
```

Comment:
I'd add `[docs here](https://………)` to all of these.

Reply:
I've added a link to the rustdocs that are on crates.io only for ethabi in commit 46b5e94a9f87df04c271cc0c5940d6775e99f134

Reply:
Based on discussion with @tomusdrw in order to be able to add publish documentation links to crates.io for the other internal toolchains evmbin, ethstore, ethkey, and whisper, we first need to fix and publish their dependencies. Should we create a separate issue that covers this?

Reply:
I've crated the new issue here: https://github.com/paritytech/parity-ethereum/issues/10739